enarx-keep-sev-shim/src/hostcall.rs

```diff
@@ -136,14 +136,16 @@ impl<'a> HostCall<'a> {
     pub unsafe fn write(&mut self, fd: usize, bytes: &[u8]) -> Result<libc::c_int, libc::c_int> {
         let bytes = &bytes[..core::cmp::min(bytes.len(), Block::buf_capacity())];
 
-        let buf_address = Address::from(self.0.buf.as_mut_ptr());
+        let mut cursor = self.0.cursor();
+        let buf = cursor.copy_slice(bytes).or(Err(libc::ENOMEM))?;
```

Comment:
Is there a reason we prefer `ENOMEM` here but `EMSGSIZE` in SGX?

Reply:
Good catch! This needs to be `EMSGSIZE`.